aten/src/ATen/native/native_functions.yaml

```diff
@@ -922,10 +922,14 @@
 - func: logical_not(Tensor self) -> Tensor
   device_check: NoCheck   # TensorIterator
   variants: function, method
+  dispatch:
+    CompositeExplicitAutograd: logical_not
```

Comment:
You also need to add a backward formula that says the input is not differentiable, right?

e.g. https://github.com/pytorch/pytorch/blob/aa2163eba53036d998d564138e10f18634c890b9/tools/autograd/derivatives.yaml#L2417-L2419

Reply:
ah, right. thanks!

Reply:
I feel like we should do this by default, this is a common thing folks miss

Reply:
Do you mean to add this because we don't want to raise an error if the inputs require grad? Is it possible for it to not raise an error here even without the formula, because the output is bool... i.e, the number of differentiable outputs is 0.

Reply:
Without this PR, logical_not does not raise an error when passed inputs that require_grad:
```
>>> import torch
>>> x = torch.randn(3, requires_grad=True)
>>> torch.logical_not(x)
tensor([False, False, False])
```
So we probably want to replicate that behavior, and to replicate that behavior the way I know of is to add the backward formula (otherwise, the operator errors out, right?)

Reply:
Ah yeah you are definitely right here. Codegen has no way of checking the dtype of the output, so it can only rely on output_differentiability, which can only exist if the formula exists. More generally though the logic is slightly more involved.
For example `new_zeros` is composite, but doesn't have a formula, and shouldn't error because it is listed in `DONT_REQUIRE_DERIVATIVE`.